component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -721,4 +726,14 @@ private void setUserClaimValues(AuthenticatedUser authenticatedUser, String user
 			throw new AuthenticationFailedException(errorMessage, e);
 		}
 	}
+
+	private String buildAbsoluteURL(String redirectUrl) throws URISyntaxException, URLBuilderException {
+
+		URI uri = new URI(redirectUrl);
+		if (uri.isAbsolute()) {
+			return redirectUrl;
+		} else {
+			return ServiceURLBuilder.create().addPath(redirectUrl).build().getAbsolutePublicURL();
+		}
```

Comment:
else block can be removed as follows.
```suggestion
		if (uri.isAbsolute()) {
			return redirectUrl;
		}
		return ServiceURLBuilder.create().addPath(redirectUrl).build().getAbsolutePublicURL();
```

Reply:
Fixed